clients/fides-js/src/lib/consent.tsx

```diff
@@ -0,0 +1,209 @@
+const validateBannerOptions = (config: FidesConfig): boolean => {
+  // Check if options is an invalid type
+  if (config.options === undefined || typeof config.options !== "object") {
+    return false;
+  }
+  // todo- more validation here?
+
+  if (!config.options.privacyCenterUrl) {
+    debugLog(
+      config.options.debug,
+      "Invalid banner options: privacyCenterUrl is required!"
+    );
+    return false;
+  }
+
+  if (config.options.privacyCenterUrl) {
+    try {
+      // eslint-disable-next-line no-new
+      new URL(config.options.privacyCenterUrl);
+    } catch (e) {
+      debugLog(
+        config.options.debug,
+        "Invalid banner options: privacyCenterUrl is an invalid URL!",
+        config
+      );
+      return false;
+    }
+  }
+
+  return true;
+};
+
+/**
+ * Construct user geolocation str to be ingested by Fides API
+ * Returns null if geolocation cannot be constructed by provided params
+ */
+const constructLocation = (
+  debug: boolean,
+  geoLocation: UserGeolocation
+): string | null => {
+  debugLog(debug, "validating getLocation...");
+  if (geoLocation.location) {
+    return geoLocation.location;
+  }
+  if (geoLocation.country && geoLocation.region) {
+    return `${geoLocation.country}-${geoLocation.region}`;
+  }
+  debugLog(
+    debug,
+    "cannot construct user location from provided geoLocation params..."
+  );
+  return null;
+};
+
+/**
+ * Fetch the user's geolocation from an external API
+ */
+const getLocation = async (options: FidesOptions): Promise<UserGeolocation> => {
+  debugLog(options.debug, "Running getLocation...");
+  const { geolocationApiUrl } = options;
+
+  if (!geolocationApiUrl) {
+    debugLog(
+      options.debug,
+      "Location cannot be found due to no configured geoLocationApiUrl."
+    );
+    return {};
+  }
+
+  debugLog(
+    options.debug,
+    `Calling geolocation API: GET ${geolocationApiUrl}...`
+  );
+  const fetchOptions: RequestInit = {
+    mode: "cors",
+  };
+  const response = await fetch(geolocationApiUrl, fetchOptions);
+
+  if (!response.ok) {
+    debugLog(
+      options.debug,
+      "Error getting location from geolocation API, returning {}. Response:",
+      response
+    );
+    return {};
+  }
+
+  try {
+    const body = await response.json();
+    debugLog(
+      options.debug,
+      "Got location response from geolocation API, returning:",
+      body
+    );
+    return body;
+  } catch (e) {
+    debugLog(
+      options.debug,
+      "Error parsing response body from geolocation API, returning {}. Response:",
+      response
+    );
+    return {};
+  }
+};
+
+/**
+ * Initialize the Fides Consent overlay components.
+ *
+ * Includes fetching location and experience, if not provided, setting up banner and modal,
+ * showing/hiding links, and showing/hiding the banner.
+ *
+ * (see the type definition of ConsentBannerOptions for what options are available)
+ */
+export const initOverlay = async (config: FidesConfig): Promise<void> => {
```

Comment:
Similar to the comment on the `App` component, passing through the whole config object feels like a code smell - we're basically just passing around a huge global variable when we do that, instead of defining sharp boundaries. It also shows up in this function: you end up doing a hundred little `config.options`, `config.experience`, `config.geolocation` calls.

I'm pushing a lil' refactor of this now to show you what I mean 👍

Reply:
Done, take a look...

Reply:
thank you! This is a lot cleaner

Reply:
✨